http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProvider.scala

```diff
@@ -422,6 +428,54 @@ class SamAuthProvider[F[_]: OpenTelemetryMetrics](
       }
     } yield token
 
+  /**
+   * Looks up the workspace parent Sam resource for the given google project.
+   *
+   * This method is used to populate workspaceId for Leonardo resources created
+   * with the v1 routes, which are in terms of google project. Once Leonardo clients
+   * are migrated to the v2 routes this method can be removed.
+   *
+   * Although we expect all google projects to have workspace parents in Sam, this
+   * method will not fail if a workspace cannot be retrieved. Logs and metrics are
+   * emitted for successful and failed workspace retrievals.
+   *
+   * @param userInfo the user info containing an access token
+   * @param googleProject the google project whose workspace parent to look up
+   * @param ev trace id
+   * @return optional workspace ID
+   */
+  override def lookupWorkspaceParentForGoogleProject(userInfo: UserInfo, googleProject: GoogleProject)(implicit
```

Comment:
Here's where it looks up the workspaceId from Sam. This code is shared for runtimes, disks, and apps.

Reply:
Gotcha, so for GCP, Rawls creates the workspace and then calls Sam to create the corresponding resource? So we are more or less guaranteed that the Sam resource has been created by the time Leo makes this call?

Reply:
Yup that's correct, there should always be a workspace Sam resource for any Terra workspace in use today. Even so I tried to make this code handle errors and emit logs/metrics, just in case something goes wrong -- we don't want this code to cause runtime creation failures.